src/x11.cc

```diff
@@ -258,7 +258,7 @@ static int x11_error_handler(Display *d, XErrorEvent *err) {
     const char *minor = xcb_errors_get_name_for_minor_code(
         xcb_errors_ctx, err->request_code, err->minor_code);
     if (minor != nullptr) {
-      const std::size_t size = strlen(base_name) + strlen(extension) + 4;
+      const std::size_t size = strlen(major) + strlen(minor) + 4;
```

Comment:
This is what was causing the crash. Not sure why it's `snprintf` fails over `code_description` being too small (isn't that the point of snprintf - to truncate?).